### ISSUE TITLE
Expose override in qhub-config.yaml for prefect

### ISSUE THREAD

**Issue by costrouc** [Quansight/qhub#799]
## Description

Currently the prefect chart provided in QHub is too restrictive. Would like to provide the option for more flexible Prefect deployments while making the helm chart follow prefect more closely.

## Acceptance Criteria

 - [x] add an overrides variable to prefect chart that follows the model in [jupyterhub]( https://github.com/Quansight/qhub/tree/main/qhub/template/%7B%7B%20cookiecutter.repo_directory%20%7D%7D/infrastructure/modules/kubernetes/services/jupyterhub)
 - add an overrides key to the `qhub-config.yaml` in prefect that allows user based overrides

```yaml
prefect:
   enabled: true
   ...
   overrides:
       ...
````

**Comment by viniciusdc** [comment]
Tested it and got the following resuts:

```
# module.prefect.helm_release.prefect will be created
# module.prefect.helm_release.prefect will be created
   + resource "helm_release" "prefect" {
       + atomic                     = false
       + chart                      = "modules/kubernetes/services/prefect/chart"
       + cleanup_on_fail            = false
       + create_namespace           = false
       + dependency_update          = false
       + disable_crd_hooks          = false
       + disable_openapi_validation = false
       + disable_webhooks           = false
       + force_update               = false
       + id                         = (known after apply)
       + lint                       = false
       + manifest                   = (known after apply)
       + max_history                = 0
       + metadata                   = (known after apply)
       + name                       = "prefect"
       + namespace                  = "dev"
       + recreate_pods              = false
       + render_subchart_notes      = true
       + replace                    = false
       + reset_values               = false
       + reuse_values               = false
       + skip_crds                  = false
       + status                     = "deployed"
       + timeout                    = 300
       + values                     = [
           + <<-EOT
                 spec:
                   template:
                     spec:
                       containers:
                         - env:
                             - name: IMAGE_PULL_SECRETS
                               value: ''
                             - name: PREFECT__CLOUD__AGENT__LABELS
                               value: '[]'
                             - name: JOB_MEM_REQUEST
                               value: ''
                             - name: JOB_MEM_LIMIT
                               value: ''
                             - name: JOB_CPU_REQUEST
                               value: ''
                             - name: JOB_CPU_LIMIT
                               value: ''
                             - name: IMAGE_PULL_POLICY
                               value: ''
                             - name: PREFECT__BACKEND
                               value: cloud
                             - name: PREFECT__CLOUD__AGENT__AGENT_ADDRESS
                               value: http://:8080
             EOT,
           + "[{'PREFECT__CLOUD__AGENT__LABELS': ['dev', 'prod'], 'JOB_MEM_REQUEST': '20Gi'}]",
         ]
       + verify                     = false
       + version                    = "0.1.0"
       + wait                       = true
       + wait_for_jobs              = false
 
       + set {
           + name  = "cloudApi"
           + value = "https://api.prefect.io"
         }
       + set {
           + name  = "namespace"
           + value = "dev"
         }
       + set {
           + name  = "prefectImage"
           + value = "prefecthq/prefect:0.14.22-python3.8"
         }
 
       + set_sensitive {
           # At least one attribute in this block is (or was) sensitive,
           # so its contents will not be displayed.
         }
     }

```
Tying to solve the folowing error:
```
Error: ---> error unmarshaling JSON: while decoding JSON: json: cannot unmarshal array into Go value of type map[string]interface {} [{'PREFECT__CLOUD__AGENT__LABELS': ['dev', 'prod'], 'JOB_MEM_REQUEST': '20Gi'}]
 │ 
 │   with module.prefect.helm_release.prefect,
 │   on modules/kubernetes/services/prefect/main.tf line 1, in resource "helm_release" "prefect":
 │    1: resource "helm_release" "prefect" {
 │ 
```

**Comment by viniciusdc** [comment]
Error solved and deplymen was suecessuf, simple test on qhub-config:

```yaml
prefect:
  enabled: true
  overrides:
      prefect_agent:
        prefectLabels: [dev, prod]
        job:
          resources:
            requests:
              memory: 20Gi
```
Output on Prefec-agent service (by K9s):
```yaml
Environment:                                                                                                                                                                                                                                                                                                                                                                                       
│       PREFECT__CLOUD__AGENT__LABELS:         ["dev","prod"]                                                                                                                                                                       │
│       JOB_MEM_REQUEST:                       20Gi                                                                                                                                                                                 
```

**Comment by viniciusdc** [comment]
https://github.com/Quansight/qhub/pull/813

**Comment by viniciusdc** [comment]
Closed by #813
